### PR TITLE
Implement Audio constructor

### LINKED ISSUE #21929: Implement Audio constructor
Spec:
https://html.spec.whatwg.org/multipage/media.html#the-audio-element (see NamedConstructor usage in WebIDL)
https://html.spec.whatwg.org/multipage/media.html#dom-audio

Code: `components/script/dom/webidls/HTMLAudioElement.webidl`, `components/script/dom/htmlaudioelement.rs`

Tests:
* `./mach test-wpt https://github.com/servo/servo/tree/master/tests/wpt/web-platform-tests/html/semantics/embedded-content/the-audio-element`
* `./mach test-wpt tests/wpt/web-platform-tests/html/semantics/embedded-content/media-elements`
* `./mach test-wpt tests/wpt/web-platform-tests/html/dom/interfaces.https.html`
* (see https://github.com/servo/servo/blob/master/tests/wpt/README.md#updating-test-expectations for dealing with unexpected results)

Now that we support actual audio output, we should implement the missing constructor that is used by many websites to create an HTMLAudioElement.

### PR DESCRIPTION
**Pull request by vn-ki**
Implement Audio constructor

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #21929 (github issue number if applicable).

- [ ] There are tests for these changes OR
- [x] These changes do not require tests because *tests were already there and the expectations were updated*.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/22227)

### CODE REVIEW

File: components/script/dom/htmlaudioelement.rs [hunk 1]
```diff
@@ -40,4 +46,19 @@ impl HTMLAudioElement {
             HTMLAudioElementBinding::Wrap,
         )
     }
+
+    pub fn Audio(
```
Comment (jdm) [inline components/script/dom/htmlaudioelement.rs]: Please add a link to https://html.spec.whatwg.org/multipage/media.html#dom-audio here.

**Review by jdm — CHANGES_REQUESTED** [review]
It's nice to see a bunch of tests passing!

File: components/script/dom/htmlaudioelement.rs [hunk 2]
```diff
@@ -40,4 +46,19 @@ impl HTMLAudioElement {
...
+        window: &Window,
+        src: Option<DOMString>,
+    ) -> Fallible<DomRoot<HTMLAudioElement>> {
+        let document = window.Document();
+        let audio = HTMLAudioElement::new(local_name!("audio"), None, &document);
+
+        audio.upcast::<HTMLMediaElement>().SetPreload(DOMString::from("auto"));
```
[the remark this answers is not in the document]
Comment (vn-ki) [reply components/script/dom/htmlaudioelement.rs]: How should I handle the `Result` that is returned?

[the remark this answers is not in the document]
Comment (jdm) [reply components/script/dom/htmlaudioelement.rs]: You can use `.expect("should be infallible")` for each of them.

File: tests/wpt/metadata/html/dom/interfaces.https.html.ini [hunk 3]
```diff
@@ -6993,18 +6915,9 @@
   [HTMLMediaElement interface: new Audio() must inherit property "ended" with the proper type]
     expected: FAIL
 
-  [HTMLMediaElement interface: new Audio() must inherit property "autoplay" with the proper type]
-    expected: FAIL
-
   [HTMLMediaElement interface: new Audio() must inherit property "loop" with the proper type]
     expected: FAIL
```
Comment (vn-ki) [inline tests/wpt/metadata/html/dom/interfaces.https.html.ini]: The expectations of these tests did not change. I do not know why?

File: tests/wpt/metadata/html/semantics/embedded-content/media-elements/playing-the-media-resource/pause-remove-from-document.html.ini [hunk 4]
```diff
@@ -1,5 +1,6 @@
 [pause-remove-from-document.html]
   type: testharness
+  expected: TIMEOUT
   [paused state when removing from a document]
-    expected: FAIL
+    expected: TIMEOUT
```
Comment (vn-ki) [inline tests/wpt/metadata/html/semantics/embedded-content/media-elements/playing-the-media-resource/pause-remove-from-document.html.ini]: Is this a mistake from my part? This tests was timing out every time I ran it.

Comment (jdm) [reply tests/wpt/metadata/html/semantics/embedded-content/media-elements/playing-the-media-resource/pause-remove-from-document.html.ini]: It means that the more of the test now executes than it did before.

File: tests/wpt/metadata/html/dom/interfaces.https.html.ini (hunk 3 above)
Comment (jdm) [reply tests/wpt/metadata/html/dom/interfaces.https.html.ini]: We don't support those properties yet.

**Pull request merged**